### PR TITLE
Refactor diff_drive demo

### PR DESCRIPTION
**Pull request by caguero**
Refactors the `diff_drive` demo leveraging the latest improvements in `ros_gz`. 

* I replaced the previous launch file from Python to XML.
* QoS settings are also used.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: ros_gz_sim_demos/README.md [hunk 1]
```diff
@@ -56,22 +56,12 @@ Trigger the camera:
...
 
-    ros2 launch ros_gz_sim_demos diff_drive.launch.py
+    ros2 launch ros_gz_sim_demos diff_drive.launch.xml
 
-Then unpause and send a command
+Then send a command.
 
     ros2 topic pub /model/vehicle_blue/cmd_vel geometry_msgs/msg/Twist "{linear: {x: 5.0}, angular: {z: 0.5}}"
 
-This demo also shows the use of custom QoS parameters. The commands are
```
Comment (ahcorde) [inline ros_gz_sim_demos/README.md]: why do you remove this comment?

**Review by ahcorde — CHANGES_REQUESTED** [review]
we are not publishing the `tf` from Gazebo. 

My suggestion here is to add the tf to the bridge
```yaml
- ros_topic_name: "tf"
  gz_topic_name: "/model/vehicle_green/tf"
  ros_type_name: "tf2_msgs/msg/TFMessage"
  gz_type_name: "gz.msgs.Pose_V"
  direction: GZ_TO_ROS
- ros_topic_name: "tf"
  gz_topic_name: "/model/vehicle_blue/tf"
  ros_type_name: "tf2_msgs/msg/TFMessage"
  gz_type_name: "gz.msgs.Pose_V"
  direction: GZ_TO_ROS
```

and a static transform for each vehicle:
```xml
<node pkg="tf2_ros" exec="static_transform_publisher" args="--frame-id world  --child-frame-id vehicle_green/odom" />
<node pkg="tf2_ros" exec="static_transform_publisher" args="--frame-id world  --child-frame-id vehicle_blue/odom" />
```

And modify the RViz file to sue the "Fixed frame" as `world` and add the other odometry marker for the green vehicle

File: ros_gz_sim_demos/README.md (hunk 1 above)
Comment (caguero) [reply ros_gz_sim_demos/README.md]: The latest version of the code ([79ece9e](https://github.com/gazebosim/ros_gz/pull/635/commits/79ece9ed0babb7869ede938120f8cd324400e34b)) preserves that block.

**Comment by caguero** [comment]
> 80d120b



> we are not publishing the `tf` from Gazebo.
> 
> My suggestion here is to add the tf to the bridge
> 
> ```yaml
> - ros_topic_name: "tf"
>   gz_topic_name: "/model/vehicle_green/tf"
>   ros_type_name: "tf2_msgs/msg/TFMessage"
>   gz_type_name: "gz.msgs.Pose_V"
>   direction: GZ_TO_ROS
> - ros_topic_name: "tf"
>   gz_topic_name: "/model/vehicle_blue/tf"
>   ros_type_name: "tf2_msgs/msg/TFMessage"
>   gz_type_name: "gz.msgs.Pose_V"
>   direction: GZ_TO_ROS
> ```
> 
> and a static transform for each vehicle:
> 
> ```
> <node pkg="tf2_ros" exec="static_transform_publisher" args="--frame-id world  --child-frame-id vehicle_green/odom" />
> <node pkg="tf2_ros" exec="static_transform_publisher" args="--frame-id world  --child-frame-id vehicle_blue/odom" />
> ```
> 
> And modify the RViz file to sue the "Fixed frame" as `world` and add the other odometry marker for the green vehicle

Thanks for the suggestion. Added in [80d120b](https://github.com/gazebosim/ros_gz/pull/635/commits/80d120bbe80228557e3dc2bc820c3205b1842d75). Could you give it a try please?

**Review by ahcorde — CHANGES_REQUESTED** [review]
Added some suggestions here https://github.com/gazebosim/ros_gz/pull/676

**Review by ahcorde — APPROVED** [review]

**Pull request merged**